fdbserver/RestoreApplier.actor.cpp

```diff
@@ -88,27 +88,25 @@ ACTOR Future<Void> restoreApplierCore(RestoreApplierInterface applierInterf, int
 }
 
 // The actor may be invovked multiple times and executed async.
-// No race condition as long as we do not wait or yield when operate the shared data, it should be fine,
-// because all actors run on 1 thread.
+// No race condition as long as we do not wait or yield when operate the shared data.
+// Multiple such actors can run on different fileIDs, because mutations in different files belong to different versions;
+// Only one actor can process mutations from the same file
 ACTOR static Future<Void> handleSendMutationVectorRequest(RestoreSendMutationVectorVersionedRequest req,
                                                           Reference<RestoreApplierData> self) {
-	state int numMutations = 0;
+	if (self->processedFileState.find(req.fileIndex) == self->processedFileState.end()) {
+		self->processedFileState.insert(std::make_pair(req.fileIndex, NotifiedVersion(0)));
+	}
+	state std::map<uint32_t, NotifiedVersion>::iterator curFileState = self->processedFileState.find(req.fileIndex);
```

Comment:
These four lines are equivalent to `NotifiedVersion& nv = self->processedFileState[req.fileIndex];` , because map's [] default initializes the value if it's not already inserted, and NotifiedVersion has a default constructor which initializes the Version to 0.